src/rosdistro/manifest_provider/git.py

```diff
@@ -47,51 +45,39 @@
 def git_manifest_provider(_dist_name, repo, pkg_name):
     assert repo.version
     try:
-        release_tag = get_release_tag(repo, pkg_name)
+        release_tag = repo.get_release_tag(pkg_name)
         package_xml = _get_package_xml(repo.url, release_tag)
         return package_xml
     except Exception as e:
         raise RuntimeError('Unable to fetch package.xml: %s' % e)
 
 
-def _git_version_gte(version):
-    global _git_client_version
-    if not _git_client_version:
-        cmd = [_git_client_executable, '--version']
-        result = _run_command(cmd)
-        _git_client_version = result['output'].split()[-1]
-    return LooseVersion(_git_client_version) >= LooseVersion(version)
-
-
 def _get_package_xml(url, tag):
     base = tempfile.mkdtemp('rosdistro')
     try:
-        assert _git_client_executable is not None, "'git' not found"
-        if _git_version_gte('1.8.0'):
+        git = Git(base)
+        if git.version_gte('1.8.0'):
             # Directly clone the required tag with least amount of additional history. This behaviour
             # has been available since git 1.8.0:
             # https://git.kernel.org/cgit/git/git.git/tree/Documentation/git-clone.txt?h=v1.8.0#n158
-            cmd = [_git_client_executable, 'clone', url, base, '--depth', '1', '--branch', tag]
-            result = _run_command(cmd, base)
+            result = git.command('clone', url, base, '--depth', '1', '--branch', tag)
             if result['returncode'] != 0:
                 raise RuntimeError('Could not clone repository "%s" at tag "%s"' % (url, tag))
         else:
             # Old git doesn't support cloning a tag directly, so check it out after a full clone.
-            cmd = [_git_client_executable, 'clone', url, base]
-            result = _run_command(cmd, base)
+            git = Git(base)
```

Comment:
Why create a new instance? Can't the following code use the instance created in line 58?